pkg/steps/multi_stage/init.go

```diff
@@ -88,58 +84,53 @@ func (s *multiStageTestStep) createCredentials(ctx context.Context) error {
 // the individual steps are run to make sure the appropriate
 // SecretProviderClasses already exist and are available for the test pods.
 func (s *multiStageTestStep) createSPCs(ctx context.Context) error {
-	toCreate := map[string]*csiapi.SecretProviderClass{}
+	spcsToCreate := map[string]*csiapi.SecretProviderClass{}
 
+	logrus.Infof("Creating SPCs for actual credential usage...")
+	// Create grouped SPCs for actual credential usage
+	for _, step := range append(s.pre, append(s.test, s.post...)...) {
+		collectionMountGroups := groupCredentialsByCollectionAndMountPath(step.Credentials)
+
+		for _, credentials := range collectionMountGroups {
+			spcName := getSPCName(s.jobSpec.Namespace(), credentials[0].Collection, credentials[0].MountPath, credentials)
+			secrets, err := buildGCPSecretsParameter(credentials)
+			if err != nil {
+				return fmt.Errorf("could not marshal secrets for mount path %s: %w", credentials[0].MountPath, err)
+			}
+			spcsToCreate[spcName] = buildSecretProviderClass(spcName, s.jobSpec.Namespace(), secrets)
+		}
+	}
+
+	// Create SPCs for sidecar censoring; each credential gets its own SPC.
+	logrus.Infof("Creating SPCs for sidecar censoring...")
+	seenCredentials := make(map[string]bool)
 	for _, step := range append(s.pre, append(s.test, s.post...)...) {
 		for _, credential := range step.Credentials {
-			name := fmt.Sprintf("%s-%s-spc", s.jobSpec.Namespace(), credential.Name)
-			if _, exists := toCreate[name]; exists {
+			if seenCredentials[credential.Name] {
```

Comment:
I've noticed that this check happens in several places. If we forbid two credentials with the same name, wouldn't it be appropriate to add this policy to the code that validates a ci-operator configuration? [pkg/validation/config.go](https://github.com/openshift/ci-tools/blob/c291f916d7c3da878c0fc59ea3574d82acf7b81b/pkg/validation/config.go)

Reply:
This check is for the scenario when there's more steps in a workflow which use the same credentials. E.g. this [workflow](https://steps.ci.openshift.org/workflow/stackrox-automation-flavors-aks-e2e) -- has steps [stackrox-stackrox-begin](https://steps.ci.openshift.org/reference/stackrox-stackrox-begin) and [stackrox-stackrox-end](https://steps.ci.openshift.org/reference/stackrox-stackrox-end), and both have the `stackrox-automation-flavors` credential in their definitions.